docs/orchestrating-elastic-stack-applications/maps.asciidoc

```diff
@@ -0,0 +1,248 @@
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  config:
+     logging.level: debug
+----
+
+Alternatively, settings can be provided via a Secret specified in the `configRef` element:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  elasticsearchRef:
+    name: elasticsearch-sample
+  configRef:
+    secretName: maps-config
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: maps-config
+stringData:
+  elastic-maps-server.yml: |-
+    logging.level: debug
+----
+Refer to <<{p}-compute-resources-kibana-and-apm>> for adjusting compute resources for {ems}
+
+[id="{p}-maps-scaling"]
+=== Scale out an {ems} deployment
+
+To deploy more than one instance of maps, all the instances must mount the data volume containing the basemap read only. Once that is the case scaling out is just a matter of increasing the `count` attribute.
+
+[id="{p}-maps-http-configuration"]
+== HTTP configuration
+
+experimental[]
+
+[id="{p}-maps-http-publish"]
+=== Load balancer settings and TLS SANs
+
+By default a `ClusterIP` link:https://kubernetes.io/docs/concepts/services-networking/service/[service] is created and associated with the {ems} deployment.
+If you want to expose maps externally with a link:https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer[load balancer], it is recommended to include a custom DNS name or IP in the self-generated certificate.
+
+Refer to <<{p}-static-ip-custom-domain>> for more details.
+
+[id="{p}-maps-http-custom-tls"]
+=== Provide your own certificate
+
+If you want to use your own certificate, the required configuration is identical to Elasticsearch. See <<{p}-custom-http-certificate>>.
+
+[id="{p}-maps-http-disable-tls"]
+=== Disable TLS
+
+You can disable the generation of the self-signed certificate and hence disable TLS. See <<{p}-disable-tls>>.
+
+
+[id="{p}-maps-ingress"]
+==== Ingress and Kibana configuration
+In order to use {ems} from your Kibana instances you need to configure Kibana to fetch maps from your {ems} instance by using the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server-kibana[`map.emsUrl`] configuration key. The value of this setting needs to be the URL where the {ems} instance is reachable from your browser. In order to work correctly the certificates presented by {ems} need to be trusted by the browser and the URL should have the same origin as the URL where your Kibana is hosted to avoid cross origin resource issues.  Please see the link:{eck_github}/tree/{eck_release_branch}/config/recipes/[recipe section] for an example how to set this up using an Ingress resource.
```

Comment:
```suggestion
To use {ems} from your Kibana instances, you need to configure Kibana to fetch maps from your {ems} instance by using the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server-kibana[`map.emsUrl`] configuration key. The value of this setting needs to be the URL where the {ems} instance is reachable from your browser. The certificates presented by {ems} need to be trusted by the browser, and the URL must have the same origin as the URL where your Kibana is hosted to avoid cross origin resource issues. See the link:{eck_github}/tree/{eck_release_branch}/config/recipes/[recipe section] for an example on how to set this up using an Ingress resource.
```